templates/telemetry/block-autoplay/block-autoplay.1.schema.json

```diff
@@ -0,0 +1,163 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+	"title": "block-autoplay",
+	"properties": {
+		"id": {
+			"type": "integer",
+			"minimum": 0
+		},
+		@TELEMETRY_CLIENTID_1_JSON@,
+		"branch": {
+			"type": "string",
+			"enum": [
+        "control",
+        "allow-and-notRemember",
+        "deny-and-notRemember",
+        "allow-and-remember",
+        "deny-and-remember"
+      ]
+		},
+		"type": {
```

Comment:
This still allows a type of "counts" with an array of "settingsChanged".

Reply:
> I still can't understand how to use "oneOf" to make the types mapping to their correspond objects

Yeah, it is very difficult to create a 100% accurate schema for this JSON.  If the object was always named the same i.e. 'payload' then oneOf is easier to apply (in that case type would be in the payload).  Without restructuring the data you would have to allow additional properties in which case any of the objects could exist; you could verify that if type == 'counts' a counters object exists but nothing would prevent you from also having a settings object and passing validation.